src/views/Register.js

```diff
@@ -0,0 +1,182 @@
+  const [data, setData] = useState([])
+  const { username, email, password } = data
+  const [error, setError] = useState('none')
+
+  const validateForm = () => {
+    return data.email && data.password
+  }
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+
+    const variables = {
+      input: {
+        username: username,
+        email: email,
+        password: password
+      }
+    }
+
+    mutate(mutation, variables)
+      .then(() => { history.push('/login') })
+      .catch((e) => {
+        const message = e[0].message.substring(96)
+        setError(message)
+      })
+  }
+
+  return (
+    <>
+      <Container>
+        <ErrorMessage error={error}>{error}</ErrorMessage>
+        <RegisterWrapper>
+          <Title>Register</Title>
+          <Form onSubmit={handleSubmit}>
+            <Form.Group size='lg' controlId='email'>
+              <Form.Label>Username</Form.Label>
+              <Form.Control
+                type='text'
+                value={data.username}
+                onChange={e => setData({ ...data, username: e.target.value })}
+                className='form'
+              />
+            </Form.Group>
+
+            <Form.Group size='lg' controlId='email'>
+              <Form.Label>Email</Form.Label>
+              <Form.Control
+                type='email'
+                value={data.email}
+                onChange={e => setData({ ...data, email: e.target.value })}
+                className='form'
+              />
+            </Form.Group>
+
+            <Form.Group size='lg' controlId='password'>
+              <Form.Label>Password</Form.Label>
+              <Form.Control
+                type='password'
+                value={data.password}
+                onChange={e => setData({ ...data, password: e.target.value })}
+                className='form'
+              />
+            </Form.Group>
+
+            {validateForm()
+              ? <ButtonRegister block size='md' type='submit'>Register </ButtonRegister>
+              : <ButtonRegisterDisabled block size='md' type='submit' disabled>Register </ButtonRegisterDisabled>}
+            <ButtonLogin
+              block size='md'
+              onClick={(e) => {
+                e.preventDefault()
+                history.push('/login')
+              }}
+            >Log In
+            </ButtonLogin>
+          </Form>
+        </RegisterWrapper>
+      </Container>
+    </>
+  )
+}
+
+export default withRouter(Register)
+
+const RegisterWrapper = styled.div`
+  width: 50%;
+  margin: 45px auto;
+  background: #EEF0F2;
+  border-radius: 20px;
+  padding: 50px;
+  font-weight: 600;
+  font-family: 'Open Sans', sans-serif;
+  font-style: normal;
+  font-weight: 600;
+  font-size: 16px;
+  line-height: 22px;
+  height: 450px;
+
+  @media only screen and (max-width: 1199px){
+    width: 75%;
+  }
+
+  @media only screen and (max-width: 767px){
+    width: 96%;
+  }
+`
+
+const ButtonRegister = styled.button`
```

Comment:
This and all the code below is the duplicate code I was referring to above in Login.